tracker/src/main/java/org/matomo/sdk/extra/InstallReferrerReceiver.java

```diff
@@ -34,12 +33,13 @@ public void onReceive(Context context, Intent intent) {
             Timber.tag(TAG).d("Dropping forwarded intent");
             return;
         }
-        SharedPreferences preferences = Matomo.getInstance(context.getApplicationContext()).getPreferences();
         if (intent.getAction().equals(REFERRER_SOURCE_GPLAY)) {
             String referrer = intent.getStringExtra(ARG_KEY_GPLAY_REFERRER);
             if (referrer != null) {
-                preferences.edit().putString(PREF_KEY_INSTALL_REFERRER_EXTRAS, referrer).apply();
-                Timber.tag(TAG).d("Stored Google Play referrer extras: %s", referrer);
+                new Thread(() -> {
+                    Matomo.getInstance(context.getApplicationContext()).getPreferences().edit().putString(PREF_KEY_INSTALL_REFERRER_EXTRAS, referrer).apply();
+                    Timber.tag(TAG).d("Stored Google Play referrer extras: %s", referrer);
+                }).start();
```

Comment:
**If** we move this action off the broadcast receivers thread, we should use `goAsync()`, because we might get undefined behavior, if the broadcastreceiver finishes execution before our thread is finished.